src/tag.rs

```diff
@@ -18,17 +18,24 @@ use sealed::sealed;
 pub trait Ext {
     /// Evaluates this [`TagOperation`] for the given `tags`.
     #[must_use]
-    fn eval(&self, tags: &[String]) -> bool;
+    fn eval<I, S>(&self, tags: I) -> bool
+    where
+        S: AsRef<str>,
+        I: Iterator<Item = S> + Clone;
```

Comment:
In APIs it's almost always better to accept `IntoIterator` instead just `Iterator`.